pkg/operator/operator.go

```diff
@@ -187,216 +206,157 @@ func (o *operator) syncHTTPEndpoint(ctx context.Context) func(obj interface{}) {
 	}
 }
 
-func (o *operator) loadCertChain(ctx context.Context) (*credentials.CertChain, error) {
-	log.Info("Getting TLS certificates")
-
-	watchCtx, watchCancel := context.WithTimeout(ctx, time.Minute)
-	defer watchCancel()
-	fsevent := make(chan struct{})
-	fserr := make(chan error)
-
-	go func() {
-		log.Infof("Starting watch for certs on filesystem: %s", o.config.Credentials.Path())
-		err := fswatcher.Watch(watchCtx, o.config.Credentials.Path(), fsevent)
-		// Watch always returns an error, which is context.Canceled if everything went well
-		if err != nil {
-			if !errors.Is(err, context.Canceled) {
-				// Ignore context.Canceled
-				fserr <- fmt.Errorf("error starting watch on filesystem: %w", err)
-			} else {
-				fserr <- nil
-			}
-
-			return
-		}
-
-		close(fsevent)
-	}()
-
-	var certChain *credentials.CertChain
-	for {
-		chain, err := credentials.LoadFromDisk(o.config.Credentials.RootCertPath(), o.config.Credentials.CertPath(), o.config.Credentials.KeyPath())
-		if err == nil {
-			log.Info("TLS certificates loaded successfully")
-			watchCancel()
-			certChain = chain
-			break
-		}
-		log.Infof("TLS certificate not found; waiting for disk changes. err=%v", err)
-		select {
-		case <-fsevent:
-			log.Debug("Watcher found activity on filesystem")
-			continue
-		case <-watchCtx.Done():
-			return nil, errors.New("timeout while waiting to load TLS certificates")
-		}
-	}
-
-	return certChain, <-fserr
-}
-
 func (o *operator) Run(ctx context.Context) error {
 	log.Info("Dapr Operator is starting")
 	healthzServer := health.NewServer(log)
 
-	err := o.mgr.Add(nonLeaderRunnable{func(ctx context.Context) error {
-		// start healthz server
-		if rErr := healthzServer.Run(ctx, healthzPort); rErr != nil {
-			return fmt.Errorf("failed to start healthz server: %w", rErr)
+	/*
+		Make sure to set `ENABLE_WEBHOOKS=false` when we run locally.
+	*/
+	if !strings.EqualFold(os.Getenv("ENABLE_WEBHOOKS"), "false") {
```

Comment:
The current behavior is anti-pattern. An unset variable should be interpreted as false (so as ENABLE_WEBHOOK=false), while in this case it's interpreted as true.

This change impacts local development only so it's non-breaking.

```suggestion
	// Make sure to set `DISABLE_WEBHOOKS=true` when we run locally.
	if !utils.IsTruthy(os.Getenv("DISABLE_WEBHOOKS")) {
```

Reply:
I don't find any reference to ENABLE_WEBHOOKS in dapr/dapr or dapr/docs. My recommendation is not to bundle unrelated changes to the same PR. It allows easier troubleshooting if it adds issues. I am OK keeping the non-ideal handling of this value and have it as a separate PR.